docs/notebooks/phrases-optimization-benchmark.ipynb

```diff
@@ -0,0 +1,373 @@
+   "source": [
+    "! python setup.py install\n",
+    "%autoreload "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2017-06-29 13:13:36,521 : INFO : collecting all words and their counts\n",
+      "2017-06-29 13:13:36,524 : INFO : PROGRESS: at sentence #0, processed 0 words and 0 word types\n",
+      "2017-06-29 13:14:09,283 : INFO : collected 4400410 word types from a corpus of 17003506 words (unigram + bigrams) and 1701 sentences\n",
+      "2017-06-29 13:14:09,284 : INFO : using 4400410 counts as vocab in Phrases<0 vocab, min_count=5, threshold=10.0, max_vocab_size=40000000>\n",
+      "2017-06-29 13:14:09,385 : INFO : collecting all words and their counts\n",
+      "2017-06-29 13:14:09,387 : INFO : PROGRESS: at sentence #0, processed 0 words and 0 word types\n",
+      "2017-06-29 13:14:41,861 : INFO : collected 4400410 word types from a corpus of 17003506 words (unigram + bigrams) and 1701 sentences\n",
+      "2017-06-29 13:14:41,863 : INFO : using 4400410 counts as vocab in Phrases<0 vocab, min_count=5, threshold=10.0, max_vocab_size=40000000>\n",
+      "2017-06-29 13:14:41,974 : INFO : collecting all words and their counts\n",
+      "2017-06-29 13:14:41,976 : INFO : PROGRESS: at sentence #0, processed 0 words and 0 word types\n",
+      "2017-06-29 13:15:15,134 : INFO : collected 4400410 word types from a corpus of 17003506 words (unigram + bigrams) and 1701 sentences\n",
+      "2017-06-29 13:15:15,135 : INFO : using 4400410 counts as vocab in Phrases<0 vocab, min_count=5, threshold=10.0, max_vocab_size=40000000>\n",
+      "2017-06-29 13:15:15,238 : INFO : collecting all words and their counts\n",
+      "2017-06-29 13:15:15,240 : INFO : PROGRESS: at sentence #0, processed 0 words and 0 word types\n",
+      "2017-06-29 13:15:54,512 : INFO : collected 4400410 word types from a corpus of 17003506 words (unigram + bigrams) and 1701 sentences\n",
+      "2017-06-29 13:15:54,513 : INFO : using 4400410 counts as vocab in Phrases<0 vocab, min_count=5, threshold=10.0, max_vocab_size=40000000>\n",
+      "2017-06-29 13:15:54,612 : INFO : collecting all words and their counts\n",
+      "2017-06-29 13:15:54,615 : INFO : PROGRESS: at sentence #0, processed 0 words and 0 word types\n",
+      "2017-06-29 13:16:30,985 : INFO : collected 4400410 word types from a corpus of 17003506 words (unigram + bigrams) and 1701 sentences\n",
+      "2017-06-29 13:16:30,986 : INFO : using 4400410 counts as vocab in Phrases<0 vocab, min_count=5, threshold=10.0, max_vocab_size=40000000>\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1 loop, best of 3: 33.2 s per loop\n"
+     ]
+    }
+   ],
+   "source": [
+    "# currently on develop --- original code\n",
+    "from gensim.models import Phrases\n",
+    "bigram = Phrases(Text8Corpus(text8_file))\n",
```

Comment:
Better convert the streamed iterable to an in-memory list (using `list()`), so we don't have to iterate over the file on disk every time.

This will make the benchmark conclusions stronger (less noise and delays from other, unrelated parts of code).